src/core/MIGRATION.md

```diff
@@ -1056,15 +1056,14 @@ import { setup, start } from '../core_plugins/embeddables/public/legacy';
 import { setup, start } from '../core_plugins/visualizations/public/legacy';
 ```
 
-| Legacy Platform                                        | New Platform                               | Notes                                                                                                                              |
-|--------------------------------------------------------|--------------------------------------------|------------------------------------------------------------------------------------------------------------------------------------|
-| `core_plugins/interpreter`                             | `data.expressions`                         | still in progress                                                                                                                  |
-| `import 'ui/apply_filters'`                            | `data.filter.loadLegacyDirectives`         | `loadLegacyDirectives()` should be called explicitly where you previously relied on importing for side effects                     |
-| `import 'ui/filter_bar'`                               | `data.filter.loadLegacyDirectives`         | `loadLegacyDirectives()` should be called explicitly where you previously relied on importing for side effects                     |
-| `import 'ui/query_bar'`                                | `data.query.loadLegacyDirectives`          | `loadLegacyDirectives()` should be called explicitly where you previously relied on importing for side effects                     |
-| `import 'ui/search_bar'`                               | `data.search.loadLegacyDirectives`         | `loadLegacyDirectives()` should be called explicitly where you previously relied on importing for side effects                     |
-| `import { QueryBar } from 'ui/query_bar'`              | `data.query.ui.QueryBar`                   | --                                                                                                                                 |
-| `import { SearchBar } from 'ui/search_bar'`            | `data.search.ui.SearchBar`                 | --                                                                                                                                 |
+| Legacy Platform                                        | New Platform                               | Notes                                                                                                                                  |
+|--------------------------------------------------------|--------------------------------------------|----------------------------------------------------------------------------------------------------------------------------------------|
+| `import 'ui/apply_filters'`                            | `import { ApplyFiltersPopover } from 'data plugin'`         | `import 'data plugin/legacy` should be called to load legacy directives                                               |
```

Comment:
For static imports like these, maybe clearer to put `import { ApplyFiltersPopover } from '../data/public'` or `import { SearchBar } from '../kibana_react/public'`?

That way we don't have to differentiate between what's in `core_plugins` or `plugins`, and it feels clearer than saying 'data plugin' because we are specifying that it's available on the client.

For runtime contracts I think it still makes sense to keep `data.x.y.z`